src/modules/signer/signer.service.ts

```diff
@@ -71,10 +75,13 @@ export class SignerService {
             await this.closeConnection();
             await this.init();
         };
-        this.on(ProviderEvent.AccountChanged, accChangedHandler);
-        this.on(ProviderEvent.NetworkChanged, accChangedHandler);
-        this.on(ProviderEvent.Disconnected, this.closeConnection());
-        this.on(ProviderEvent.SessionUpdate, accChangedHandler);
+        if (this._providerType === ProviderType.MetaMask) {
+            this.on(ProviderEvent.AccountChanged, accChangedHandler);
+            this.on(ProviderEvent.NetworkChanged, accChangedHandler);
+        } else if (this._providerType === ProviderType.WalletConnect) {
+            this.on(ProviderEvent.SessionUpdate, accChangedHandler);
+            this.on(ProviderEvent.Disconnected, this.closeConnection());
```

Comment:
Shouldn't be like this?
```suggestion
            this.on(ProviderEvent.Disconnected, this.closeConnection);
```